nimble/controller/src/ble_ll_scan.c

```diff
@@ -2238,6 +2299,12 @@ ble_ll_scan_rx_isr_end(struct os_mbuf *rxpdu, uint8_t crcok)
     if (rc) {
         ble_ll_state_set(BLE_LL_STATE_STANDBY);
     }
+#if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
+    /* On handover lets increase ref count */
+    if (ble_hdr->rxinfo.user_data) {
+        ble_ll_scan_aux_data_ref(ble_hdr->rxinfo.user_data);
```

Comment:
I think you should ref/unref at the same place where copying takes place, otherwise it will be just hard to follow and prone to errors

Reply:
I would keep it here as this is single exit place from this function and we know that aux_data will be handover to LL.
In opposite there is no single place in this functions where aux_data is created/copied. If you look from the top of this function you will see that there is at least 2 paths to get to the end of this function with valid aux_data, so we have to do decide to ref in those 2 places  - I think that would complicate code. Single ref in the end of this function looks much better for me.

Reply:
imo the point of using refs is that you ref immediately when making copy, this way you always know how many refs you have and you can't make mistake here. copying reference in one place and increasing ref count in other place makes this hard to follow, especially that here it's based on some assumption about contents of `ble_hdr->rxinfo.user_data` which for me is counter-intuitive, but I guess it's ok